frontend/src/presentation/pages/QuestionsPage.tsx

```diff
@@ -1,30 +1,30 @@
 import React, { useState, useEffect } from "react";
-import { Row, Col, Breadcrumb, Button, Spin, Alert } from "antd";
+import { Row, Col, Breadcrumb, Button, Spin, Alert, Modal } from "antd";
 import { QuestionList } from "../components/QuestionList";
 import { QuestionDetail } from "../components/QuestionDetail";
 import { LandingComponent } from "../components/LandingComponent";
+import { NewQuestionForm } from "../components/NewQuestionForm/NewQuestionForm";
 import { Question } from "../../domain/entities/Question";
 import { QUESTIONS_PAGE_TEXT } from "presentation/utils/constants";
 import { questionUseCases } from "../../domain/usecases/QuestionUseCases";
 import styles from "./QuestionsPage.module.css";
 import { ROUTES, ERRORS } from "presentation/utils/constants";
 import { handleError } from "presentation/utils/errorHandler";
 import { AddQuestionButton } from "presentation/components/buttons/AddQuestionButton";
-import { useSearchParams, useNavigate } from "react-router-dom"; // Import hooks from react-router-dom
+import { useSearchParams } from "react-router-dom";
 
 const QuestionsPage: React.FC = () => {
     const [questions, setQuestions] = useState<Question[]>([]);
     const [selectedQuestion, setSelectedQuestion] = useState<Question | null>(null);
     const [isLoading, setIsLoading] = useState(true);
     const [isQuestionLoading, setIsQuestionLoading] = useState(false);
     const [error, setError] = useState<string | null>(null);
+    const [isModalVisible, setIsModalVisible] = useState(false);
 
-    const [searchParams, setSearchParams] = useSearchParams(); // Use this to handle query parameters
-    const navigate = useNavigate(); // Use this for programmatic navigation
+    const [searchParams, setSearchParams] = useSearchParams();
 
-    const selectedQuestionId = searchParams.get('selected'); // Get the selected question ID from query parameters
+    const selectedQuestionId = searchParams.get('selected');
```

Comment:
Hmm, sounds like we're using query params? If so, maybe `?code` sounds better than `?selected`? I think this is fine too though